### DISCUSSION TITLE
Balancing currency exchange

### DISCUSSION THREAD

**Discussion by mtsz-pl** [ananthakumaran/paisa#147] [Q&A] [answered]
When I create this record:
```
2019/09/21 transaction
    Assets:A  -1000 PLN
    Assets:B  1000 PLN
```

I get expected zero balance:
```
                   0  Assets
           -1000 PLN    A
            1000 PLN    B
--------------------
                   0
```

but when I replace deduction's currency
```
2019/09/21 transaction
    Assets:A  -250 USD @ 4 PLN
    Assets:B  1000 PLN
```

it doesn't balance anymore
```
            1000 PLN
            -250 USD  Assets
            -250 USD    A
            1000 PLN    B
--------------------
            1000 PLN
            -250 USD
```
Is there something I am missing? Or is it desired behaviour? The problem is I have plenty of exchange transactions, and it assumulates a lot in the balance

**Comment by ananthakumaran** [comment, marked as answer]
This is expected and this question comes up on [reddit](https://www.reddit.com/r/plaintextaccounting/comments/i8vgzv/ledgercli_foreign_transaction_balance_confusion/) frequently. What happens is expected.

> The problem is I have plenty of exchange transactions, and it assumulates a lot in the balance

Do you use the assets > balance page to check your balance? there you should get the balance in your default currency.

**Reply by mtsz-pl** [reply]
I understand, thank you. "Assets > Balance" has correctly calculated values in default currency
